locust/test/test_dispatch.py

```diff
@@ -3291,6 +3291,241 @@ class User3(User):
         self.assertEqual(_user_count_on_worker(dispatched_users, worker_nodes[2].id), 3)
 
 
+class TestRampUpUsersFromZeroWithFixed(unittest.TestCase):
+    class Case:
+        def __init__(self, fixed_counts: Tuple[int], weights: Tuple[int], target_user_count: int):
+            self.fixed_counts = fixed_counts
+            self.weights = weights
+            self.target_user_count = target_user_count
+
+    def case_handler(self, cases: List[Case], expected: Dict[str, int], user_classes=List[User]):
+        self.assertEqual(len(cases), len(expected))
+
+        for case_num in range(len(cases)):
+            # Reset to defaul values
+            for user_class in user_classes:
+                user_class.weight, user_class.fixed_count = 1, 0
+
+            case = cases[case_num]
+            self.assertEqual(
+                len(case.fixed_counts) + len(case.weights),
+                len(user_classes),
+                msg="Invalid test case or user list.",
+            )
+
+            fixed_users_list = user_classes[: len(case.fixed_counts)]
+            weighted_users_list = user_classes[len(case.fixed_counts) :]
+
+            for user, fixed_count in zip(fixed_users_list, case.fixed_counts):
+                user.fixed_count = fixed_count
+
+            for user, weight in zip(weighted_users_list, case.weights):
+                user.weight = weight
+
+            worker_node1 = WorkerNode("1")
+
+            users_dispatcher = UsersDispatcher(worker_nodes=[worker_node1], user_classes=user_classes)
+            users_dispatcher.new_dispatch(target_user_count=case.target_user_count, spawn_rate=0.5)
+            users_dispatcher._wait_between_dispatch = 0
+
+            iterations = list(users_dispatcher)
+            self.assertDictEqual(iterations[-1]["1"], expected[case_num])
+
+    def test_ramp_up_2_weigted_user_with_1_fixed_user(self):
+        class User1(User):
+            ...
+
+        class User2(User):
+            ...
+
+        class User3(User):
+            ...
+
+        self.case_handler(
+            cases=[
+                self.Case(fixed_counts=(), weights=(1, 1, 1), target_user_count=3),
```

Comment:
This case makes the test's name (`test_ramp_up_2_weigted_user_with_1_fixed_user`) a little misleading, no?

Reply:
Yep, move this case to **test_ramp_up_various_count_weigted_and_fixed_users**